x-pack/solutions/observability/plugins/observability/public/pages/alert_details/hooks/related_alerts/use_build_related_alerts_query.ts

```diff
@@ -0,0 +1,181 @@
+import {
+  ALERT_END,
+  ALERT_GROUP,
+  ALERT_INSTANCE_ID,
+  ALERT_RULE_TAGS,
+  ALERT_RULE_UUID,
+  ALERT_START,
+  ALERT_STATUS,
+  ALERT_UUID,
+} from '@kbn/rule-data-utils';
+import dedent from 'dedent';
+import moment from 'moment';
+import { ObservabilityFields } from '../../../../../common/utils/alerting/types';
+import { TopAlert } from '../../../../typings/alerts';
+
+interface Props {
+  alert: TopAlert<ObservabilityFields>;
+}
+
+export function useBuildRelatedAlertsQuery({ alert }: Props): QueryDslQueryContainer {
+  const groups = alert.fields[ALERT_GROUP];
+  const shouldGroups =
+    groups?.map(({ field, value }) => ({
+      bool: {
+        boost: 2.0,
+        must: [
+          { term: { 'kibana.alert.group.field': field } },
+          { term: { 'kibana.alert.group.value': value } },
+        ],
+      },
+    })) ?? [];
+  const shouldRule = alert.fields[ALERT_RULE_UUID]
+    ? [
+        {
+          term: {
+            'kibana.alert.rule.uuid': {
+              value: alert.fields[ALERT_RULE_UUID],
+              boost: 1.0,
+            },
+          },
+        },
+      ]
+    : [];
+  const startDate = moment(alert.fields[ALERT_START]);
+  const endDate = alert.fields[ALERT_END] ? moment(alert.fields[ALERT_END]) : undefined;
+  const tags = alert.fields[ALERT_RULE_TAGS] ?? [];
+  const instanceId = alert.fields[ALERT_INSTANCE_ID]?.split(',') ?? [];
+
+  return {
+    bool: {
+      filter: [
+        {
+          range: {
+            [ALERT_START]: {
+              gte: startDate.clone().subtract(1, 'days').toISOString(),
+              lte: startDate.clone().add(1, 'days').toISOString(),
+            },
+          },
+        },
+      ],
+      must_not: [
+        {
+          term: {
+            [ALERT_UUID]: {
+              value: alert.fields[ALERT_UUID],
+            },
+          },
+        },
+      ],
+      should: [
+        ...shouldGroups,
+        ...shouldRule,
+        {
+          term: {
+            [ALERT_STATUS]: {
+              value: alert.fields[ALERT_STATUS],
+              boost: 2,
+            },
+          },
+        },
+        {
+          function_score: {
+            functions: [
+              {
+                exp: {
+                  [ALERT_START]: {
+                    origin: startDate.toISOString(),
+                    scale: '10m',
+                    offset: '10m',
+                    decay: 0.5,
+                  },
+                },
+                weight: 10,
+              },
+              ...(endDate
+                ? [
+                    {
+                      exp: {
+                        [ALERT_END]: {
+                          origin: endDate.toISOString(),
+                          scale: '10m',
+                          offset: '10m',
+                          decay: 0.5,
+                        },
+                      },
+                      weight: 10,
+                    },
+                  ]
+                : []),
```

Comment:
I feel like exponential decay is reasonable. I think this should be based on the interval distance though (so might need script score). Specifically, if alerts have interval intervals $[a_1, b_1]$ and $[a_2, b_2]$ I would use something like
```math
score = \exp(-(\max(a_1 - b_2, 0) - \max(a_2 - b_1, 0)) / T)
```
This calculates the distance between the intervals if they don't overlap and is otherwise 0.

One efficiency grounds one should probably add some filter conditions as well, i.e. must not end time of one alert << start time of other. This will allow Lucene to skip many candidate matches and only run the script scoring on the remainder.